File: Sheet 6 (Complexity Analysis)/K.cpp
#include <iostream>

using namespace std;

int main()
{
    bool p(true);
    int n, x, a, b;
    cin >> n >> x;
    while (n--)
    {
        cin >> a >> b;
        if (x == a || x == b || x == 7 - a || x == 7 - b)
        {
            p = false;
        }
        x = 7 - x;
    }
    cout << (p ? "YES" : "NO") << endl;
    return 0;
}

File: Sheet 6 (Complexity Analysis)/K.py
n=int(input())
x=int(input())
top=x
bottom=7-top
sides=[tuple(map(int,input().split())) for _ in range(n)]
for a,b in sides:
    if a==top or a==bottom or b==top or b==bottom:
        print("NO")
        break
else:
    print("YES")
